tests/testsuite/freshness.rs

```diff
@@ -492,62 +508,84 @@ fn changing_bin_features_caches_targets() {
         .build();
 
     p.cargo("build")
-        .with_stderr(
-            "\
-[COMPILING] foo v0.0.1 ([..])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
-",
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
+        .run();
+    p.rename_run("foo", "off1")
+        .with_stdout_data(str![[r#"
+feature off
+
+"#]])
         .run();
-    p.rename_run("foo", "off1").with_stdout("feature off").run();
 
     p.cargo("build --features foo")
-        .with_stderr(
-            "\
-[COMPILING] foo v0.0.1 ([..])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
-",
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
+        .run();
+    p.rename_run("foo", "on1")
+        .with_stdout_data(str![[r#"
+feature on
+
+"#]])
         .run();
-    p.rename_run("foo", "on1").with_stdout("feature on").run();
 
     /* Targets should be cached from the first build */
 
     let mut e = p.cargo("build -v");
 
     // MSVC does not include hash in binary filename, so it gets recompiled.
     if cfg!(target_env = "msvc") {
-        e.with_stderr(
-            "\
-[DIRTY] foo v0.0.1 ([..]): the list of features changed
-[COMPILING] foo[..]
-[RUNNING] `rustc [..]
-[FINISHED] `dev`[..]",
-        );
+        e.with_stderr_data(str![[r#"
+[DIRTY] foo v0.0.1 ([ROOT]/foo): the dependency foo was rebuilt ([DIRTY_REASON_NEW_TIME], [DIRTY_REASON_DIFF] after last build at [DIRTY_REASON_OLD_TIME])
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
```

Comment:
Any insight from how this happened?  Overall, we should be using `SNAPSHOTS=overwrite` so it seems like this shouldn't have happened so I'm wondering if you were doing some things manually (and if so, if there was a reason) or if there was a bug somewhere.

Reply:
I mostly use (or at least start from) `SNAPSHOTS=overwrite`, but for these `if cfg!(target_env = "msvc") {...}` blocks I had to do manual edit because I didn't have a Windows machine to run it locally 😅.

Reply:
Oh, yeah, that is a pain :)